lib/NonlinearSolveFirstOrder/src/solve.jl

```diff
@@ -206,13 +208,17 @@ function SciMLBase.__init(
             prob, alg, u, fu, J, du; kwargs...
         )
 
-        return GeneralizedFirstOrderAlgorithmCache(
+        cache = GeneralizedFirstOrderAlgorithmCache(
             fu, u, u_cache, prob.p, du, J, alg, prob, globalization,
             jac_cache, descent_cache, linesearch_cache, trustregion_cache,
             stats, 0, maxiters, maxtime, alg.max_shrink_times, timer,
-            0.0, true, termination_cache, trace, ReturnCode.Default, false, kwargs
+            0.0, true, termination_cache, trace, ReturnCode.Default, false, kwargs,
+            initializealg
         )
     end
+
+    NonlinearSolveBase.run_initialization!(cache)
```

Comment:
Should this be inside the `@static_timeit`?

Reply:
yes